src/platform/plugins/shared/dashboard/server/dashboard_saved_object/dashboard_saved_object.ts

```diff
@@ -85,11 +86,46 @@ export const createDashboardSavedObjectType = ({
         create: dashboardAttributesSchemaV3,
       },
     },
+    4: {
+      // deprecate **all** non-indexed fields from the mapping so that we can treat the dashboard SO as a black box
+      changes: [
+        {
+          type: 'mappings_deprecation',
+          deprecatedMappings: [
+            'hits',
+            'kibanaSavedObjectMeta',
+            'optionsJSON',
+            'panelsJSON',
+            'sections',
+            'refreshInterval',
+            'controlGroupInput',
+            'timeFrom',
+            'timeRestore',
+            'timeTo',
+          ],
+        },
+      ],
+      schemas: {
+        forwardCompatibility: dashboardAttributesSchemaV4.extends({}, { unknowns: 'ignore' }),
+        create: dashboardAttributesSchemaV4,
+      },
```

Comment:
As a follow up to https://github.com/elastic/kibana/pull/222900/files#r2136480651, if mapping removal is **will not** be allowed, do we even need a new version to change the schema? What is the benefit of deprecating mappings?